src/MoBi.Core/Serialization/ORM/Mappers/ProjectMetaDataToProjectMapper.cs

```diff
@@ -91,9 +92,22 @@ private void addNonProjectBuildingBlocks(List<IBuildingBlock> moduleBuildingBloc
          var simulationSettingsBlocks = moduleBuildingBlocks.OfType<SimulationSettings>().ToList();
          simulationSettingsBlocks.Each(x => moduleBuildingBlocks.Remove(x));
 
-         addModuleBuildingBlocks(moduleBuildingBlocks);
+
+         var uniqueBuildingBlockTypes = hasUniqueBuildingBlocks(moduleBuildingBlocks);
```

Comment:
you can use the method directly in the condition

`if(hasUniqueBuildingBlocks(moduleBuildingBlocks))`